homeassistant/config_entries.py

```diff
@@ -1100,13 +1100,15 @@ def async_update_entry(
 
     @callback
     def async_setup_platforms(
-        self, entry: ConfigEntry, platforms: Iterable[str]
+        self, entry: ConfigEntry, platforms: Iterable[Platform | str]
     ) -> None:
         """Forward the setup of an entry to platforms."""
         for platform in platforms:
             self.hass.async_create_task(self.async_forward_entry_setup(entry, platform))
 
-    async def async_forward_entry_setup(self, entry: ConfigEntry, domain: str) -> bool:
+    async def async_forward_entry_setup(
+        self, entry: ConfigEntry, platform: Platform | str
+    ) -> bool:
```

Comment:
I'm not sure we should change the argument name. Technically it's a breaking change.

Reply:
If it's a breaking change, then I will undo the name change.

Reply:
The function signature doesn't require the arguments to be positional only, so someone could use keywords with would change. _Just to be clear, I don't think it makes sense to add a positional only requirement for these methods._